kombu/transport/azureservicebus.py

```diff
@@ -0,0 +1,154 @@
+"""Azure Service Bus Message Queue transport.
+
+The transport can be enabled by setting the CELERY_BROKER_URL to:
+
+```
+azureservicebus://{SAS policy name}:{SAS key}@{Service Bus Namespace}
+```
+
+Note that the Shared Access Policy used to connect to Azure Service Bus
+requires Manage, Send and Listen claims since the broker will create new
+queues and delete old queues as required.
+
+Note that if the SAS key for the Service Bus account contains a slash, it will
+have to be regenerated before it can be used in the connection URL.
+
+More information about Azure Service Bus:
+https://azure.microsoft.com/en-us/services/service-bus/
+
+"""
+from __future__ import absolute_import, unicode_literals
+
+from os import getenv
+import string
+
+from kombu.five import Empty, text_t
+from kombu.utils.encoding import bytes_to_str, safe_str
+from kombu.utils.json import loads, dumps
+from kombu.utils.objects import cached_property
+
+from . import virtual
+
+try:
+    from azure.servicebus import ServiceBusService, Message, Queue
+except ImportError:
+    ServiceBusService = Message = Queue = None
+
+# dots are replaced by dash, all other punctuation replaced by underscore.
+CHARS_REPLACE_TABLE = {
+    ord(c): 0x5f for c in string.punctuation if c not in '_'
+}
+
+
+class Channel(virtual.Channel):
+    """Azure Service Bus channel."""
+
+    default_visibility_timeout = 1800  # 30 minutes.
+    domain_format = 'kombu%(vhost)s'
+    _sbq = None
+    _queue_cache = {}
+
+    def __init__(self, *args, **kwargs):
+        if ServiceBusService is None:
+            raise ImportError('Azure Service Bus transport requires the '
+                              'azure-servicebus library')
+
+        super(Channel, self).__init__(*args, **kwargs)
+
+        for queue in self.sbq.list_queues():
```

Comment:
There is an [open issue](https://github.com/celery/kombu/pull/834#issue-168161180) with AWS SQS transport, not requiring `ListQueues` permissions. In the proposed fix, a predefined list of queue names is given by configuration. Would that apply here as well?

Reply:
Azure ServiceBus has three kinds of permissions: `listen`, `send`, and `manage` ([docs](https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-sas#shared-access-authorization-policies)). Given that we're creating new queues from the connector, we'll anyways need the `manage` permissions so might as well use the list queues call here. Thoughts?